### PR TITLE
[ENH] Add support for multiple datablocks in Nihon Kohden files

### PR DESCRIPTION
**Pull request by fraimondo**
#### Reference issue
Fixes/Implements #9429.

#### What does this implement/fix?

Nihon Kohden reader was not tested using multiple datablocks (maybe due to only having continuous recordings? don't really know yet). This PR allows to read NK EEG files with multiple datablocks.

#### Additional information
Still needs to be tested against the corresponding EDF files.  PR is WIP.

### CODE REVIEW

**Comment by fraimondo** [comment]
So I tested with the NK files that were sent and this script:

```Python
from pathlib import Path
import mne
from numpy.testing import assert_array_almost_equal

fname = '/Users/fraimondo/Downloads/TEST NK/FA0755PV.EEG'
raw = mne.io.read_raw_nihon(fname, preload=True)

edf_path = Path('/Users/fraimondo/data/TEST NK/')

raws = []

fnames = sorted([x for x in edf_path.glob('FA0755PV*+.edf')])
for fname in fnames:
    print(fname.name)
    t_edf = mne.io.read_raw_edf(fname, preload=True)
    raws.append(t_edf)

raw_edf = mne.concatenate_raws(raws)
raw_edf.drop_channels('Events/Markers')


assert_array_almost_equal(raw._data, raw_edf._data)
```

I tested recording FA0755PV (4 datablocks) and FA07548P (2 datablocks). No assertion errors.

**Comment by agramfort** [comment]
still WIP @fraimondo ?

**Comment by fraimondo** [comment]
Not any more. Ready for review.

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -49,6 +49,8 @@ Enhancements
 
 - Add support for interpolating oxy and deoxyhaemoglobin data types (:gh:`9431` by `Robert Luke`_)
 
+- Add support for multiple datablocks (acquistions with pauses) in `mne.io.read_raw_nihon` (:gh:`9437` by `Fede Raimondo`_)
```
Comment (agramfort) [inline doc/changes/latest.inc:52]:
```suggestion
- Add support for multiple datablocks (acquistions with pauses) in :func:`mne.io.read_raw_nihon` (:gh:`9437` by `Fede Raimondo`_)
```

**Review by agramfort — COMMENTED** [review]
I understand it's different blocks of data? If so would it make sense to add BAD edge annotations as done here:

https://github.com/mne-tools/mne-python/blob/main/mne/io/base.py#L1692

File: doc/changes/latest.inc (hunk 1 above)
Comment (fraimondo) [reply doc/changes/latest.inc]: It's already adding the two events exactly as you say: `BAD boundary` and `EDGE boundary`.

**Review by agramfort — APPROVED** [review]

File: mne/io/nihon/tests/test_nihon.py [hunk 2]
```diff
@@ -57,5 +57,5 @@ def test_nihon_eeg():
 
     msg = 'No LOG file exists. Annotations will not be read'
     with pytest.warns(RuntimeWarning, match=msg):
-        annot = _read_nihon_annotations(bad_fname, orig_time=None)
-        assert annot is None
+        annot = _read_nihon_annotations(bad_fname)
+        assert all(len(x) == 0 for _, x in annot.items())
```
Comment (larsoner) [inline mne/io/nihon/tests/test_nihon.py:61]:
```suggestion
        assert all(len(x) == 0 for x in annot.values())
```

**Review by larsoner — APPROVED** [review]
Looks nice and clean, +1 for merge even without my tiny suggestion!

File: mne/io/nihon/tests/test_nihon.py (hunk 2 above)
Comment (fraimondo) [reply mne/io/nihon/tests/test_nihon.py]: With the suggestion is even cleaner!

Nothing more to add, PR can be closed.

**Comment by larsoner** [comment]
Thanks @fraimondo !

**Pull request merged**